src/cryptonote_core/blockchain.cpp

```diff
@@ -1803,6 +1803,27 @@ bool Blockchain::handle_alternative_block(const block& b, const crypto::hash& id
     }
     bei.cumulative_difficulty += current_diff;
 
+    bei.block_cumulative_weight = cryptonote::get_transaction_weight(b.miner_tx);
+    for (const crypto::hash &txid: b.tx_hashes)
+    {
+      cryptonote::tx_memory_pool::tx_details td;
+      if (!m_tx_pool.get_transaction_info(txid, td))
+      {
+        MERROR_VER("Block with id: " << epee::string_tools::pod_to_hex(id) << " (as alternative) has unknown transaction hash " << txid << ".");
+        bvc.m_verifivation_failed = true;
+        return false;
+      }
+      bei.block_cumulative_weight += td.weight;
+    }
+    if (alt_chain.size())
+    {
+      bei.block_cumulative_weight += prev_data.cumulative_weight;
+    }
+    else
+    {
+      bei.block_cumulative_weight += m_db->get_block_weight(m_db->get_block_height(b.prev_id));
+    }
```

Comment:
As I read `Blockchain::create_block_template()`, my understanding of the role of the `bei.block_cumulative_weight` (and `data.cumulative_weight`) field is to serve as the rough estimate of the current block weight median:

```c++
    // cheat and use the weight of the block we start from, virtually certain to be acceptable
    // and use 1.9 times rather than 2 times so we're even more sure
    if (parent_in_main)
    {
      median_weight = m_db->get_block_weight(height - 1);
      already_generated_coins = m_db->get_block_already_generated_coins(height - 1);
    }
    else
    {
      median_weight = prev_data.cumulative_weight - prev_data.cumulative_weight / 20;
      already_generated_coins = alt_chain.back().already_generated_coins;
    }
```

So, having it as an accumulation of the weights of all blocks in an alt chain as done here seems wrong to me. Am I missing something?

Reply:
Yes, I misread, I thought this was subtracting the parent block from the top block, but it just removes 1/20. This code is a bodge anyway, this should really hold the actual median.